pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+ * [ RHSM_VALID, RHSM_EXPIRED, RHSM_WARNING, RHN_CLASSIC, RHSM_PARTIALLY_VALID, RHSM_REGISTRATION_REQUIRED ]
+ */
+var subscriptionStatusValues = [
+    'RHSM_VALID',
+    'RHSM_EXPIRED',
+    'RHSM_WARNING',
+    'RHN_CLASSIC',
+    'RHSM_PARTIALLY_VALID',
+    'RHSM_REGISTRATION_REQUIRED'
+];
+var requestUpdate = function() {
+    service.call(
+        '/EntitlementStatus',
+        'com.redhat.SubscriptionManager.EntitlementStatus',
+        'check_status',
+        [])
+        .done(function(result) {
+            client.subscriptionStatus.serviceStatus = subscriptionStatusValues[result[0]];
+            updateTimeout = null;
+            client.getSubscriptionStatus();
+        })
+        .catch(function(ex) {
+            client.subscriptionStatus.serviceStatus = undefined;
+            updateTimeout = null;
+            console.warn("EntitlementStatus.check_status() failed:", ex);
+        });
+
+    /* TODO: Don't use a timeout here. Needs better API */
+    updateTimeout = window.setTimeout(statusUpdateFailed, 30000);
+};
+
+function processStatusOutput(text, exitDetails) {
+    if (exitDetails.problem === 'access-denied') {
+        client.subscriptionStatus.status = "access-denied";
+        needRender();
+        return;
+    }
+    /* if output isn't as expected, maybe not properly installed? */
+    if (text.indexOf('Overall Status:') === -1) {
+        console.warn(text, exitDetails);
+        client.subscriptionStatus.status = "not-found";
+        return;
+    }
+
+    /* clear old subscription details */
+    client.subscriptionStatus.products = [];
+
+    var status = parseSingleSubscription(text);
+    client.subscriptionStatus.status = status['Overall Status'];
+
+    needRender();
+
+    /* if refresh was requested, try again - otherwise get details */
+    if (client.subscriptionStatus !== 'Unknown')
+        getSubscriptionDetails();
+}
+
+var gettingStatus = false;
+var getStatusRequested = false;
+/* get subscription summary using 'subscription-manager status'*/
+client.getSubscriptionStatus = function() {
+    if (gettingStatus) {
+        getStatusRequested = true;
+        return;
+    }
+    getStatusRequested = false;
+    gettingStatus = true;
+    /* we need a buffer for 'subscription-manager status' output, since that can fail with a return code != 0
+     * even if we need its output (no valid subscription)
+     */
+    var status_buffer = '';
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     *
+     * 'subscription-manager status' will only return with exit code 0 if all is well (and subscriptions current)
+     */
+    cockpit.spawn(['subscription-manager', 'status'],
+            { directory: '/', superuser: "try", environ: ['LC_ALL=C'], err: "out" })
+        .stream(function(text) {
+            status_buffer += text;
+            return text.length;
```

Comment:
I don't think this line is necessary. Not returning anything has the same result.